contrib/backporting/cherry-pick

```diff
@@ -7,7 +7,7 @@ require_linux
 
 ORG=${ORG:-"cilium"}
 REPO=${REPO:-"cilium"}
-BRANCH=${BRANCH:-"master"}
+BRANCH=${BRANCH:-"main"}
```

Comment:
I think we should already be fine as we already handle this as part of the stable branching process (e.g. workflow files get the `main`/`master` references replaced with the stable branch name), and I don't think there's anything outside of that that requires additional changes following the rename of `master` to `main`.

Reply:
I think Julian is highlighting that his workflow is to use the `cherry-pick` script from an older branch which doesn't yet have these changes. Personally I always use the `contrib/` scripts from `main` branch (using `git worktree` to share the git repo with multiple copies per branch), but I appreciate that others have other workflows.

This will work with `master` for the next three months or so unless we either address this in all older branches or extend the `master` mirror longer. Bumping the cherry-pick script is simple enough, I can submit a PR for that. If you find any other broken workflows in future, feel free to send the PRs to fix those in the target branches.